arc-mlir/src/arc-dialect.cpp

```diff
@@ -37,6 +38,91 @@ ArcDialect::ArcDialect(mlir::MLIRContext *ctx) : mlir::Dialect("arc", ctx) {
 #define GET_OP_LIST
 #include "arc/ops.cpp.inc"
       >();
+  addTypes<arc::ArcTypes::AppenderType>();
+}
+
+//===----------------------------------------------------------------------===//
+// Arc Type Parser
+//===----------------------------------------------------------------------===//
+
+bool isValueType(Type type) {
+  switch (type.getKind()) {
+  case StandardTypes::BF16:
+  case StandardTypes::F16:
+  case StandardTypes::F32:
+  case StandardTypes::F64:
+  case StandardTypes::Integer:
+  case StandardTypes::Vector:
+  case StandardTypes::RankedTensor:
+  case StandardTypes::UnrankedTensor:
+  case StandardTypes::UnrankedMemRef:
+  case StandardTypes::MemRef:
+  case StandardTypes::Complex:
+  case StandardTypes::Tuple:
+    for (auto t : type.cast<TupleType>().getTypes()) {
```

Comment:
`type.cast<TupleType>().getTypes()` can't be right, none of the fallthroughs above are tuples, the cast will assert.

Reply:
```patch
diff --git a/arc-mlir/src/arc-dialect.cpp b/arc-mlir/src/arc-dialect.cpp
index cd8f967..8964665 100644
--- a/arc-mlir/src/arc-dialect.cpp
+++ b/arc-mlir/src/arc-dialect.cpp
@@ -58,15 +58,12 @@ bool isValueType(Type type) {
   case StandardTypes::UnrankedMemRef:
   case StandardTypes::MemRef:
   case StandardTypes::Complex:
+  case StandardTypes::None:
+    return true;
   case StandardTypes::Tuple:
-    for (auto t : type.cast<TupleType>().getTypes()) {
-      if (!isValueType(t)) {
+    for (auto t : type.cast<TupleType>().getTypes())
+      if (!isValueType(t))
         return false;
-      }
-    }
-    return true;
-    break;
-  case StandardTypes::None:
     return true;
   default:
     return false;
```
is probably what you want.

Reply:
✅